src/libNetwork/P2PComm.cpp

```diff
@@ -555,18 +587,32 @@ void P2PComm::ConnectionAccept([[gnu::unused]] evconnlistener* listener,
         return;
     }
 
-    auto func = [cli_sock, from]() -> void {
-        HandleAcceptedConnection(cli_sock, from);
-    };
-
-    GetInstance().m_RecvPool.AddJob(func);
+    // Set up buffer event for this new connection
+    struct event_base* base = evconnlistener_get_base(listener);
+    struct bufferevent* bev = bufferevent_socket_new(
+        base, cli_sock, BEV_OPT_CLOSE_ON_FREE | BEV_OPT_DEFER_CALLBACKS);
+    bufferevent_setcb(bev, NULL, NULL, EventCallback, from);
+    bufferevent_enable(bev, EV_READ | EV_WRITE);
 }
 
 void P2PComm::StartMessagePump(uint32_t listen_port_host, Dispatcher dispatcher,
                                BroadcastListFunc broadcast_list_retriever)
 {
     LOG_MARKER();
 
+    // Launch the thread that reads messages from the send queue
+    auto funcCheckSendQueue = [this]() mutable -> void {
+        SendJob* job = NULL;
+        while (true)
+        {
+            while (m_sendQueue.pop(job))
```

Comment:
We could change while() to if(), it's equivalent

Reply:
Since they're equivalent, would prefer to keep it this way, because the inner while makes it clear that what we want is to keep popping while there is something left to pop, and the outer while makes it clear that after everything has been popped we want to try again. Using if instead of while doesn't make it as clear I think.